common/common-http/src/main/java/org/ow2/proactive/http/HttpClientBuilder.java

```diff
@@ -0,0 +1,190 @@
+
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+
+import javax.net.ssl.SSLContext;
+
+import org.ow2.proactive.web.WebProperties;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.SSLContextBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+
+
+public class HttpClientBuilder {
+
+    private static final AllowAllTrustStrategy ACCEPT_ANY_CERTIFICATE_TRUST_STRATEGY = new AllowAllTrustStrategy();
+
+    private Boolean allowAnyHostname;
+
+    private Boolean allowAnyCertificate;
+
+    private int maxConnections;
+
+    private RequestConfig requestConfig;
+
+    private boolean useContentCompression;
+
+    private boolean useSystemProperties;
+
+    public HttpClientBuilder() {
+        this.useContentCompression = true;
+    }
+
+
+    public HttpClientBuilder allowAnyHostname(boolean acceptAnyHostname) {
+        this.allowAnyHostname = acceptAnyHostname;
+        return this;
+    }
+
+    public HttpClientBuilder allowAnyCertificate(boolean acceptAnyCertificate) {
+        this.allowAnyCertificate = acceptAnyCertificate;
+        return this;
+    }
+
+    public HttpClientBuilder disableContentCompression() {
+        this.useContentCompression = false;
+        return this;
+    }
+
+    public HttpClientBuilder insecure(boolean insecure) {
+        allowAnyHostname(insecure);
+        allowAnyCertificate(insecure);
+        return this;
+    }
+
+    public HttpClientBuilder maxConnections(int maxConnections) {
+        this.maxConnections = maxConnections;
+        return this;
+    }
+
+    public HttpClientBuilder setDefaultRequestConfig(RequestConfig requestConfig) {
+        this.requestConfig = requestConfig;
+        return this;
+    }
+
+    public HttpClientBuilder useSystemProperties() {
+        return useSystemProperties(true);
+    }
+
+    public HttpClientBuilder useSystemProperties(boolean useSystemProperties) {
+        this.useSystemProperties = useSystemProperties;
+        return this;
+    }
+
+    public CloseableHttpClient build() {
+        org.apache.http.impl.client.HttpClientBuilder httpClientBuilder =
+                createHttpClientBuilder();
+
+        boolean acceptAnyCertificate = false;
+        boolean acceptAnyHostname = false;
+
+        if (useSystemProperties) {
+            httpClientBuilder.useSystemProperties();
+
+            String property = System.getProperty(WebProperties.WEB_HTTPS_ALLOW_ANY_CERTIFICATE);
+            if ("true".equalsIgnoreCase(property)) {
+                acceptAnyCertificate = true;
+            }
+
+            property = System.getProperty(WebProperties.WEB_HTTPS_ALLOW_ANY_HOSTNAME);
+            if ("true".equalsIgnoreCase(property)) {
+                acceptAnyHostname = true;
+            }
+        }
+
+        if (allowAnyHostname != null) {
```

Comment:
Field `allowAnyCertificate` is of type _Boolean_ which allows `Boolean.TRUE`, `Boolean.FALSE` and `null` as value whereas `acceptAnyCertificate` is the primitive type that allows `true` or `false` only. Distinguishing both is important to be able to know if the property was set or not and to set the desired default value accordingly.